tests/connection_test.go

```diff
@@ -35,7 +34,9 @@ const connTestTimeout = time.Minute
 // and then unsets GOOGLE_APPLICATION_CREDENTIALS. It returns a cleanup function
 // that restores the original setup.
 func removeAuthEnvVar(t *testing.T) (*oauth2.Token, string, func()) {
-	ts, err := google.DefaultTokenSource(context.Background(), sqladmin.SqlserviceAdminScope)
+	ts, err := google.DefaultTokenSource(context.Background(),
+		"https://www.googleapis.com/auth/cloud-platform",
```

Comment:
Why do we need cloud-platform here?

Reply:
Short answer: the impersonation fails without this scope. Longer answer: let me figure out if we can limit this scope still further.

Reply:
After some digging, I found that this is the required OAuth2 scope to call generateAccessToken.


See:
- https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/generateAccessToken
- https://github.com/googleapis/google-api-go-client/blob/main/iamcredentials/v1/iamcredentials-gen.go#L85